### PR TITLE
Number of infrastructure fixes

### PR DESCRIPTION
**Pull request by jaredpar**
Couple of fixes:

- Move xunit output files to the Logs folder
- Remove DevDivInsertionFiles.sln
- Add a binary log for our restore operations
- Remove the `-buildAll` option
- Get dump files when xunit encounters a stack overflow

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure, @333fred for review

**Comment by jaredpar** [comment]
The win32 unit test failed with an Editor2.UnitTests crash again. The last test to run before the crash was
    Microsoft.CodeAnalysis.Editor.UnitTests.Peek.PeekTests.TestPeekDefinitionWhenExtensionMethodFromMetadata

https://ci.dot.net/job/dotnet_roslyn/job/dev15.7.x/job/windows_release_unit32_prtest/704/

CC @333fred

**Comment by jaredpar** [comment]
retest windows_release_unit32_prtest please

File: src/Tools/Source/RunTests/ProcDumpUtil.cs [hunk 1]
```diff
@@ -63,7 +63,8 @@ internal static Process AttachProcDump(string procDumpFilePath, int processId, s
             // /accepteula command line option to automatically accept the Sysinternals license agreement.
             // -ma	Write a 'Full' dump file. Includes All the Image, Mapped and Private memory.
             // -e	Write a dump when the process encounters an unhandled exception. Include the 1 to create dump on first chance exceptions.
-            const string procDumpSwitches = "/accepteula -ma -e";
+            // -f C00000FD.STACK_OVERFLOWC Dump when a stack overflow first chance exception is encountered. 
+            const string procDumpSwitches = "/accepteula -ma -e -f C00000FD.STACK_OVERFLOW";
```
Comment (333fred) [inline src/Tools/Source/RunTests/ProcDumpUtil.cs]: Would we also be able to add System.Reflection.TargetParameterCountException to troubleshoot the DoEvents hangs we've been seeing?

Comment (jaredpar) [reply src/Tools/Source/RunTests/ProcDumpUtil.cs]: Do you know how to express that as `-f` switch? Happy to add if we can figure that out.

**Pull request merged**